### PR TITLE
Add curl command to pull karpenter cloudformation from remote repo

### PR DESCRIPTION
**Pull request by brianhammons**
Issue #, if available:

Description of changes:
Currently, the cloudformation deploy expects the karpenter cf template to be local. This change will pull the template from the remote repository, without needing to clone the full project.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/aws/README.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ This command will create IAM resources used by Karpenter. We recommend using [Cl
 ```bash
 aws cloudformation deploy \
   --stack-name Karpenter-${CLUSTER_NAME} \
-  --template-file ./docs/aws/karpenter.cloudformation.yaml \
+  --template-file $(pwd)/$(curl -fsSO https://raw.githubusercontent.com/awslabs/karpenter/v0.2.4/docs/aws/karpenter.cloudformation.yaml | echo karpenter.cloudformation.yaml)  \
```
Comment (ellistarn) [inline docs/aws/README.md:31]: I successfully ran this without the $(pwd), why is this necessary?

Comment (ellistarn) [inline docs/aws/README.md]: Would it be possible to wire this up with `mktemp -d` to avoid littering the pwd?

**Review by ellistarn — DISMISSED** [review]

**Comment by ellistarn** [comment]
Thanks for this docs improvement!

File: docs/aws/README.md [hunk 2]
```diff
@@ -28,7 +28,7 @@ This command will create IAM resources used by Karpenter. We recommend using [Cl
 ```bash
 aws cloudformation deploy \
   --stack-name Karpenter-${CLUSTER_NAME} \
-  --template-file ./docs/aws/karpenter.cloudformation.yaml \
+  --template-file $(TEMPOUT=$(mktemp) | curl -fsSL https://raw.githubusercontent.com/awslabs/karpenter/v0.2.4/docs/aws/karpenter.cloudformation.yaml > $TEMPOUT | echo $TEMPOUT)  \
```
Comment (JacobGabrielson) [inline docs/aws/README.md:31]: I could be doing something wrong but I can't get this to worry (sorry if this is a red herring). I'm wondering if this might work? 

```bash
curl -fsSL https://raw.githubusercontent.com/awslabs/karpenter/v0.2.4/docs/aws/karpenter.cloudformation.yaml \
  | aws cloudformation deploy \
    --stack-name Karpenter-${CLUSTER_NAME} \
    --template-file /dev/stdin  \
   # the rest is the same...
```

Comment (njtran) [reply docs/aws/README.md:31]: This didn't work for me @JacobGabrielson.

The following command works for me. IIUC pipes use different subshells, so the TEMPOUT variable will be saved in a different subprocess if you pipe it. You have to define it before.
```
TEMPOUT=$(mktemp)
curl -fsSL https://raw.githubusercontent.com/awslabs/karpenter/v0.2.4/docs/aws/karpenter.cloudformation.yaml > $TEMPOUT \
| aws cloudformation deploy \
  --stack-name Karpenter-${CLUSTER_NAME} \
  --template-file ${TEMPOUT}\
  --capabilities CAPABILITY_NAMED_IAM \
  --parameter-overrides ClusterName=${CLUSTER_NAME} OpenIDConnectIdentityProvider=$(aws eks describe-cluster --name ${CLUSTER_NAME} | jq -r ".cluster.identity.oidc.issuer" | cut -c9-)
```

**Review by njtran — CHANGES_REQUESTED** [review]
Thanks for this PR! Would you be able to change the `--template-file` flag?

File: docs/aws/README.md [hunk 3]
```diff
@@ -29,11 +29,13 @@ eksctl create cluster \
 ### Create IAM Resources
 This command will create IAM resources used by Karpenter. We recommend using [CloudFormation](https://aws.amazon.com/cloudformation/) and [IAM Roles for Service Accounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) (IRSA) to manage these permissions. For production use, please review and restrict these permissions for your use case.
 ```bash
-aws cloudformation deploy \
-  --stack-name Karpenter-${CLUSTER_NAME} \
-  --template-file ./docs/aws/karpenter.cloudformation.yaml \
-  --capabilities CAPABILITY_NAMED_IAM \
-  --parameter-overrides ClusterName=${CLUSTER_NAME} OpenIDConnectIdentityProvider=$(aws eks describe-cluster --name ${CLUSTER_NAME} | jq -r ".cluster.identity.oidc.issuer" | cut -c9-)
+TEMPOUT=$(mktemp)
+curl -fsSL https://raw.githubusercontent.com/awslabs/karpenter/v0.2.4/docs/aws/karpenter.cloudformation.yaml > $TEMPOUT \
```
Comment (njtran) [inline docs/aws/README.md:33]: We cut a release v0.2.5 a bit ago. Would you be able to change this link to reflect the proper version?

**Review by njtran — COMMENTED** [review]
Looks really good! Just have one nit

**Review by njtran — APPROVED** [review]
